File: frontend/src/types/rolePermission.ts
// E:\Audit\AMS\frontend\src\types\rolePermission.ts

export type RolePermission = {
  id: number;
  role_id: number;
  permission_id: number;
  is_active: boolean;

  role_name?: string | null;
  role_description?: string | null;

  permission_key?: string | null;
  resource_type?: string | null;
  resource_key?: string | null;
  action?: string | null;
  description?: string | null;

  created_by: string | null;
  updated_by: string | null;
  created_at: string;
  updated_at: string;
};

export type RolePermissionPayload = {
  role_id: number;
  permission_id: number;
};

export type RolePermissionMessageResponse = {
  message: string;
  data: RolePermission | null;
};

File: backend/app/schemas/role_permission.py
# E:\Audit\AMS\backend\app\schemas\role_permission.py

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RolePermissionCreate(BaseModel):
    role_id: int
    permission_id: int


class RolePermissionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    role_id: int
    permission_id: int
    is_active: bool

    role_name: str | None = None
    role_description: str | None = None

    permission_key: str | None = None
    resource_type: str | None = None
    resource_key: str | None = None
    action: str | None = None
    description: str | None = None

    created_by: str | None = None
    updated_by: str | None = None
    created_at: datetime
    updated_at: datetime


class RolePermissionMessageResponse(BaseModel):
    message: str
    data: RolePermissionResponse | None = None
